pkg/rtc/mediatracksubscriptions.go

```diff
@@ -110,6 +110,9 @@ func (t *MediaTrackSubscriptions) AddSubscriber(sub types.LocalParticipant, wr *
 	if err != nil {
 		return nil, err
 	}
+	if t.params.MediaTrack.IsEncrypted() {
+		downTrack.SetTrailer(sub.GetTrailer())
```

Comment:
seems like it should follow the publisher instead of subscriber.

would avoid exposing it on the Participant interface and instead it would be in the constructor if MediaTrackReceiver

Reply:
I had the first version (just in my workspace, not committed) like that. Then I did this for two reasons
- Was wondering if we would have a participant level trailer. Currently, this PR generates a room level trailer and sends it to all participants of the room on this node in `JoinResponse`. It is imaginable that we generate per participant trailer.
- More importantly, this meant with mesh, don't have to carry the trailer in participant update/track update across nodes. It would have required attaching trailer to `TrackInfo` (maybe that is a good thing to extend `TrackInfo` to include a trailer?) and sending it in `ParticipantUpdate` across the mesh.

Reply:
got it! that makes sense.

It still seems like a per-node/room construct would be sufficient? I'm not seeing the value of having a diff trailer per participant. While the flexibility is nice, it seems we can keep things simpler unless the need arises.

Reply:
Yeah, I think that will be sufficient . Participant level is just an added flexibility if we want.

But keeping it to per-node/room means that it has to be accessible at participant level. So, the participant API is needed. That constraint means that we can do participant specific trailer in case we need it.